server/src/main/java/org/opensearch/index/compositeindex/datacube/MetricStat.java

```diff
@@ -10,29 +10,59 @@
 
 import org.opensearch.common.annotation.ExperimentalApi;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
 /**
  * Supported metric types for composite index
  *
  * @opensearch.experimental
  */
 @ExperimentalApi
 public enum MetricStat {
-    COUNT("count"),
-    AVG("avg"),
-    SUM("sum"),
-    MIN("min"),
-    MAX("max");
+    COUNT("count", null),
+    SUM("sum", null),
+    MIN("min", null),
+    MAX("max", null),
+    AVG("avg", new MetricStat[] { COUNT, SUM });
 
     private final String typeName;
+    private final MetricStat[] derivedFrom;
 
-    MetricStat(String typeName) {
+    MetricStat(String typeName, MetricStat[] derivedFrom) {
         this.typeName = typeName;
+        this.derivedFrom = derivedFrom;
     }
 
     public String getTypeName() {
         return typeName;
     }
 
+    /**
+     * Return the list of metrics that this metric is derived from
+     * For example, AVG is derived from COUNT and SUM
+     */
+    public List<MetricStat> getDerivedFromMetrics() {
```

Comment:
nit: getDependentMetrics?

Reply:
derived sounds more accurate compared to dependent